### PR TITLE
Tobin gh1703 ds image transition

### PR DESCRIPTION
**Pull request by tobine**
add missing check and test

### CODE REVIEW

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -473,6 +473,22 @@ bool ValidateBarriersToImages(layer_data *device_data, VkCommandBuffer cmdBuffer
         if (!img_barrier) continue;
 
         VkImageCreateInfo *image_create_info = &(GetImageState(device_data, img_barrier->image)->createInfo);
+        // For a Depth/Stencil image both aspects MUST be set
+        if (FormatIsDepthAndStencil(image_create_info->format)) {
+            auto aspect_mask = img_barrier->subresourceRange.aspectMask;
```
Comment (jeremy-lunarg) [inline layers/buffer_validation.cpp]: auto const

File: layers/buffer_validation.cpp [hunk 2]
```diff
@@ -473,6 +473,22 @@ bool ValidateBarriersToImages(layer_data *device_data, VkCommandBuffer cmdBuffer
...
+            auto ds_mask = VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT;
```
Comment (jeremy-lunarg) [inline layers/buffer_validation.cpp]: auto const

File: layers/buffer_validation.cpp [hunk 3]
```diff
@@ -473,6 +473,22 @@ bool ValidateBarriersToImages(layer_data *device_data, VkCommandBuffer cmdBuffer
...
+            if ((aspect_mask & ds_mask) != (ds_mask)) {
+                skip |=
+                    log_msg(core_validation::GetReportData(device_data), VK_DEBUG_REPORT_ERROR_BIT_EXT,
+                            VK_DEBUG_REPORT_OBJECT_TYPE_IMAGE_EXT, reinterpret_cast<const uint64_t &>(img_barrier->image), __LINE__,
+                            VALIDATION_ERROR_00302, "DS",
```
Comment (jeremy-lunarg) [inline layers/buffer_validation.cpp]: Do we have a test for this? I didn't see one.

File: layers/buffer_validation.cpp (hunk 1 above)
Comment (tobine) [inline layers/buffer_validation.cpp]: yep

File: layers/buffer_validation.cpp (hunk 2 above)
Comment (tobine) [inline layers/buffer_validation.cpp]: yep

File: layers/buffer_validation.cpp (hunk 3 above)
Comment (tobine) [inline layers/buffer_validation.cpp]: I screwed up my rebasing and the new test I added is not correctly pushed to the PR. Fixing it up and will push shortly.

**Comment by tobine** [comment]
@jeremy-lunarg commits should be sorted now, consts added and new test included.

**Pull request merged**